itests/org.openhab.core.thing.tests/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/TimestampProfileTest.java

```diff
@@ -58,22 +59,26 @@ public void testTimestampOnChange() {
         ArgumentCaptor<State> capture = ArgumentCaptor.forClass(State.class);
         TimestampChangeProfile timestampProfile = new TimestampChangeProfile(callback);
 
+        // No existing previous state saved, the callback is first called
         timestampProfile.onStateUpdateFromItem(new DecimalType(23));
-        verify(callback, atLeastOnce()).sendUpdate(capture.capture());
+        verify(callback, times(1)).sendUpdate(capture.capture());
         State result = capture.getValue();
-
         DateTimeType changeResult = (DateTimeType) result;
+
+        waitForAssert(() -> {
+            assertTrue(ZonedDateTime.now().isAfter(changeResult.getZonedDateTime()));
```

Comment:
Since it's just one statement you can also use:

```java
        waitForAssert(() -> assertTrue(ZonedDateTime.now().isAfter(changeResult.getZonedDateTime())));
```

It prevents scrolling when using multiple `waitForAssert` statements which may happen when someone copy/pastes it one day.

Reply:
Done